resources/sshdconfig/src/parser.rs

```diff
@@ -59,67 +59,185 @@ impl SshdConfigParser {
         }
     }
 
-    fn parse_keyword_node(&mut self, keyword_node: tree_sitter::Node, input: &str, input_bytes: &[u8]) -> Result<(), SshdConfigError> {
+    fn parse_match_node(&mut self, match_node: tree_sitter::Node, input: &str, input_bytes: &[u8]) -> Result<(), SshdConfigError> {
+        let mut criteria_map = Map::new();
+        let mut cursor = match_node.walk();
+        let mut match_object = Map::new();
+
+        for child_node in match_node.named_children(&mut cursor) {
+            if child_node.is_error() {
+                return Err(SshdConfigError::ParserError(t!("parser.failedToParseNode", input = input).to_string()));
+            }
+
+            match child_node.kind() {
+                "comment" => {}
+                "criteria" => {
+                    Self::parse_match_criteria(child_node, input, input_bytes, &mut criteria_map)?;
+                }
+                "keyword" => {
+                    Self::parse_and_insert_keyword(child_node, input, input_bytes, Some(&mut match_object))?;
+                }
+                _ => {
+                    return Err(SshdConfigError::ParserError(t!("parser.unknownNodeType", node = child_node.kind()).to_string()));
+                }
+            }
+        }
+
+        // Add the match object to the main map
+        if criteria_map.is_empty() {
+            return Err(SshdConfigError::ParserError(t!("parser.missingCriteriaInMatch", input = input).to_string()));
+        }
+        match_object.insert("criteria".to_string(), Value::Object(criteria_map));
+        Self::insert_into_map(&mut self.map, "match", Value::Object(match_object), true)?;
+        Ok(())
+    }
+
+    /// Parse a single match criteria node and insert into the provided criteria map.
+    /// Example criteria node: "user alice,bob" or "address *.*.0.1"
+    /// Inserts the criterion as a key with an array value into the `criteria_map`.
+    fn parse_match_criteria(criteria_node: tree_sitter::Node, input: &str, input_bytes: &[u8], criteria_map: &mut Map<String, Value>) -> Result<(), SshdConfigError> {
+        let mut cursor = criteria_node.walk();
+        let mut key: Option<String> = None;
+        let mut values: Vec<Value> = Vec::new();
+
+        // Iterate through named children of the criteria node
+        for child in criteria_node.named_children(&mut cursor) {
+            if child.is_error() {
+                return Err(SshdConfigError::ParserError(t!("parser.failedToParseNode", input = input).to_string()));
+            }
+
+            match child.kind() {
+                "alpha" => {
+                    let Ok(text) = child.utf8_text(input_bytes) else {
+                        return Err(SshdConfigError::ParserError(t!("parser.failedToParseNode", input = input).to_string()));
+                    };
+                    key = Some(text.to_string());
+                }
+                "boolean" | "string" => {
```

Comment:
I wonder if internally we should treat `yes`/`no` booleans as actual booleans instead of strings and just render them back to `yes`/`no` when serializing?